File: snippets/write_mirror_file.mjs
import fs from "fs";
import path from "path";

/**
 * Write one file from read_live_theme_mirror_files response.
 * @param {string} root - output directory
 * @param {{ filename: string, encoding: 'text'|'base64', content: string }} file
 */
export function writeMirrorFile(root, file) {
  const dest = path.join(root, file.filename);
  fs.mkdirSync(path.dirname(dest), { recursive: true });
  if (file.encoding === "text") {
    fs.writeFileSync(dest, file.content, "utf8");
  } else {
    fs.writeFileSync(dest, Buffer.from(file.content, "base64"));
  }
}

File: snippets/write_mirror_file.py
import base64
from pathlib import Path


def write_mirror_file(root: Path, file: dict) -> None:
    """Write one file from read_live_theme_mirror_files response."""
    dest = root / file["filename"]
    dest.parent.mkdir(parents=True, exist_ok=True)
    if file["encoding"] == "text":
        dest.write_text(file["content"], encoding="utf-8")
    else:
        dest.write_bytes(base64.b64decode(file["content"]))
